meta/Meta.cpp

```diff
@@ -6434,6 +6434,55 @@ void Meta::meta_sai_on_nat_event(
     }
 }
 
+void Meta::meta_sai_on_port_host_tx_ready_change(
+                    _In_ sai_object_id_t port_id,
+                    _In_ sai_object_id_t switch_id,
+                    _In_ sai_port_host_tx_ready_status_t host_tx_ready_status)
+{
+    SWSS_LOG_ENTER();
+
+    if (!sai_metadata_get_enum_value_name(
+            &sai_metadata_enum_sai_port_host_tx_ready_status_t,
+            host_tx_ready_status))
+    {
+        SWSS_LOG_WARN("port host_tx_ready value (%d) not found in sai_port_host_tx_ready_status_t",
+                host_tx_ready_status);
+    }
+    
+    auto ot = objectTypeQuery(port_id);
+    auto valid = false;
+    
+    switch (ot)
+    {
+        case SAI_OBJECT_TYPE_PORT:
+        // case SAI_OBJECT_TYPE_BRIDGE_PORT:
+        // case SAI_OBJECT_TYPE_LAG:
+            valid = true;
+            break;
+
+        default:
+            SWSS_LOG_ERROR("port_id %s has unexpected type: %s, expected PORT", //, BRIDGE_PORT or LAG
+                    sai_serialize_object_id(port_id).c_str(),
+                    sai_serialize_object_type(ot).c_str());
+            break;
+    }
+
+    if (valid && !m_oids.objectReferenceExists(port_id))
```

Comment:
could you please help me understand why `!m_oids.objectReferenceExists(port_id)` is needed here? My understanding is like this: 
```
if port ref does not exist:
     insert port ref

if host ready key does not exist:
    create host key

```